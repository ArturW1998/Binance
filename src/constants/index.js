export const ACTION_TYPES = {
  UPDATE_MARKET_PAIRS: 'UPDATE_MARKET_PAIRS',
  SET_ACTIVE_MARKET: 'SET_ACTIVE_MARKET',
  TOGGLE_SOCKET_STREAMS: 'TOGGLE_SOCKET_STREAMS',
};

export const PATHS = {
  BASE_PATH: 'wss://stream.binance.com',
  STREAM_PATH: '/stream',
  STREAM_PARAM: 'streams=',
};

export const LISTS = {
  PAIRS_LIST: ['BNB', 'BTC', 'ETH', 'USDT'],
  DOTS_LIST: [{ id: 1 }, { id: 2 }, { id: 3 }, { id: 4 }],
  TABLE_DATA_LIST: [
    { title: 'Pair' },
    { title: 'Latest Price' },
    { title: 'Open' },
    { title: 'High' },
    { title: 'Low' },
    { title: 'Volume' },
  ],
};

export const CONFIG = {
  BASE_NAME: '/Binance/',
  LOCALSTORAGE_KEY: 'markets',
};
