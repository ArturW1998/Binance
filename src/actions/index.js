import { SET_ACTIVE_MARKET, UPDATE_MARKET_PAIRS, TOGGLE_SOCKET_STREAMS } from '../constants';

export const setActiveMarket = data => ({ type: SET_ACTIVE_MARKET, data });

export const updateMarketPairs = data => ({ type: UPDATE_MARKET_PAIRS, data });

export const toggleSocketStreams = status => ({ type: TOGGLE_SOCKET_STREAMS, status });
