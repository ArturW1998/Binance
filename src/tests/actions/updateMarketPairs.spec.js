import { UPDATE_MARKET_PAIRS } from '../../constants';
import { updateMarketPairs } from '../../actions';

describe('updateMarketPairs action', () => {
  const expectedActionUpdateMarketPairs = {
    type: UPDATE_MARKET_PAIRS,
    data: {
      BNBBTC: {},
      ETHUSDT: {},
      ENJBTC: {},
    },
  };

  const { data } = expectedActionUpdateMarketPairs;

  it('updateMarketPairs', () => {
    expect(updateMarketPairs(data)).toEqual(expectedActionUpdateMarketPairs);
  });

  it('updateMarketPairs should return an action', () => {
    expect(updateMarketPairs(data)).toMatchSnapshot();
  });
});
