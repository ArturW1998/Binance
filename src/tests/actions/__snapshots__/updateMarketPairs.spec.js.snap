// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`updateMarketPairs action updateMarketPairs should return an action 1`] = `
Object {
  "data": Object {
    "BNBBTC": Object {},
    "ENJBTC": Object {},
    "ETHUSDT": Object {},
  },
  "type": "UPDATE_MARKET_PAIRS",
}
`;
